cfme/v2v/migrations.py

```diff
@@ -166,16 +167,16 @@ def fill(self, values):
                         ...
                     }
         """
-        for cluster in values:
-            if self.cluster_selector.is_displayed:
-                self.cluster_selector.fill(cluster)
-            for mapping in values[cluster]['mappings']:
-                self.source_datastores.fill(mapping['sources'])
-                self.target_datastores.fill(mapping['target'])
-                self.add_mapping.click()
+        if values:
+            for cluster in values:
+                if self.cluster_selector.is_displayed:
+                    self.cluster_selector.fill(cluster)
+                for mapping in values[cluster]['mappings']:
+                    self.source_datastores.fill(mapping['sources'])
+                    self.target_datastores.fill(mapping['target'])
+                    self.add_mapping.click()
         was_change = not self.mappings_tree.is_empty
-        if was_change:
-            self.logger.info("Fill operation was successful, click Next.")
+        if self.next_btn.wait_displayed():
```

Comment:
**Question**: Can we check this for each page of the form? And also, wait_displayed() would be wrong,  maybe we should check `is_enabled `? Because when we don't have any mappings, it is usually greyed out.

Reply:
@kedark3 :without wait displayed , buttons are not clicked , it needs some time to display